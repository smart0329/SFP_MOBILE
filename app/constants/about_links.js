// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    TERMS_OF_SERVICE: 'https://about.mattermost.com/default-terms/',
    PRIVACY_POLICY: 'https://about.mattermost.com/default-privacy-policy/',
};
