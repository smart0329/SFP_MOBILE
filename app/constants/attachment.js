// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const MAX_ATTACHMENT_FOOTER_LENGTH = 300;
export const ATTACHMENT_ICON_HEIGHT = 48;
export const ATTACHMENT_ICON_WIDTH = 36;
