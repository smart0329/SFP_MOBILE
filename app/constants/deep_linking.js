// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    CHANNEL: 'channel',
    PERMALINK: 'permalink',
    OTHER: 'other',
};
