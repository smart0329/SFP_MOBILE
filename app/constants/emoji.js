// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const ALL_EMOJIS = 'all_emojis';
export const MAX_ALLOWED_REACTIONS = 40;