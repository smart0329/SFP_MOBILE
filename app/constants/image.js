// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const IMAGE_MAX_HEIGHT = 350;
export const IMAGE_MIN_DIMENSION = 50;
