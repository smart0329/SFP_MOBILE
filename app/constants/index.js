// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import DeepLinkTypes from './deep_linking';
import DeviceTypes from './device';
import ListTypes from './list';
import NavigationTypes from './navigation';
import ViewTypes, {UpgradeTypes} from './view';

export {
    DeepLinkTypes,
    DeviceTypes,
    ListTypes,
    NavigationTypes,
    UpgradeTypes,
    ViewTypes,
};
