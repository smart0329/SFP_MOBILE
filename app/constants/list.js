// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

const VISIBILITY_CONFIG_DEFAULTS = {
    itemVisiblePercentThreshold: 100,
    waitForInteraction: false,
};

export default {
    VISIBILITY_CONFIG_DEFAULTS,
    VISIBILITY_SCROLL_DOWN: 'down',
    VISIBILITY_SCROLL_UP: 'up',
};
