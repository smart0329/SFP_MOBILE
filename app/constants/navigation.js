// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import keyMirror from 'mattermost-redux/utils/key_mirror';

const NavigationTypes = keyMirror({
    NAVIGATION_RESET: null,
    NAVIGATION_CLOSE_MODAL: null,
    NAVIGATION_NO_TEAMS: null,
    RESTART_APP: null,
    NAVIGATION_ERROR_TEAMS: null,
    NAVIGATION_SHOW_OVERLAY: null,
    CLOSE_MAIN_SIDEBAR: null,
    MAIN_SIDEBAR_DID_CLOSE: null,
    MAIN_SIDEBAR_DID_OPEN: null,
    CLOSE_SETTINGS_SIDEBAR: null,
    BLUR_POST_TEXTBOX: null,
});

export default NavigationTypes;
