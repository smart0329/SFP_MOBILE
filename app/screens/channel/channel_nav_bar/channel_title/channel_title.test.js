// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';
import {General} from 'mattermost-redux/constants';

import ChannelTitle from './channel_title';

jest.mock('react-intl');

describe('ChannelTitle', () => {
    const baseProps = {
        displayName: 'My Channel',
        isGuest: false,
        hasGuests: false,
        canHaveSubtitle: false,
        isSelfDMChannel: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ChannelTitle {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot when isSelfDMChannel is true', () => {
        const props = {
            ...baseProps,
            displayName: 'My User',
            isSelfDMChannel: true,
        };
        const wrapper = shallow(
            <ChannelTitle {...props}/>,
            {context: {intl: {formatMessage: (intlId) => intlId.defaultMessage}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot when is DM and has guests but the teammate is not the guest (when can show subtitles)', () => {
        const props = {
            ...baseProps,
            displayName: 'Other user',
            channelType: General.DM_CHANNEL,
            isGuest: false,
            hasGuests: true,
            canHaveSubtitle: true,
        };
        const wrapper = shallow(
            <ChannelTitle {...props}/>,
            {context: {intl: {formatMessage: (intlId) => intlId.defaultMessage}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot when is DM and has guests and the teammate is the guest (when can show subtitles)', () => {
        const props = {
            ...baseProps,
            displayName: 'Other user',
            channelType: General.DM_CHANNEL,
            isGuest: true,
            hasGuests: true,
            canHaveSubtitle: true,
        };
        const wrapper = shallow(
            <ChannelTitle {...props}/>,
            {context: {intl: {formatMessage: (intlId) => intlId.defaultMessage}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
