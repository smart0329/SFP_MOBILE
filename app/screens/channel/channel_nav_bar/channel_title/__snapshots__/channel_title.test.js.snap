// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelTitle should match snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "position": "relative",
        "top": -1,
        "width": "90%",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": undefined,
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      My Channel
    </Text>
    <Icon
      allowFontScaling={false}
      name="chevron-down"
      size={12}
      style={
        Object {
          "color": undefined,
          "marginHorizontal": 5,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`ChannelTitle should match snapshot when is DM and has guests and the teammate is the guest (when can show subtitles) 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "position": "relative",
        "top": -1,
        "width": "90%",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": undefined,
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Other user
    </Text>
    <Icon
      allowFontScaling={false}
      name="chevron-down"
      size={12}
      style={
        Object {
          "color": undefined,
          "marginHorizontal": 5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "position": "relative",
        "top": -1,
        "width": "90%",
      }
    }
  >
    <FormattedText
      defaultMessage="This person is a guest"
      ellipsizeMode="tail"
      id="channel.isGuest"
      numberOfLines={1}
      style={
        Object {
          "color": undefined,
          "fontSize": 14,
          "opacity": 0.6,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`ChannelTitle should match snapshot when is DM and has guests but the teammate is not the guest (when can show subtitles) 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "position": "relative",
        "top": -1,
        "width": "90%",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": undefined,
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Other user
    </Text>
    <Icon
      allowFontScaling={false}
      name="chevron-down"
      size={12}
      style={
        Object {
          "color": undefined,
          "marginHorizontal": 5,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`ChannelTitle should match snapshot when isSelfDMChannel is true 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "position": "relative",
        "top": -1,
        "width": "90%",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": undefined,
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      <FormattedText
        defaultMessage="{displayname} (you)"
        id="channel_header.directchannel.you"
        values={
          Object {
            "displayname": "My User",
          }
        }
      />
    </Text>
    <Icon
      allowFontScaling={false}
      name="chevron-down"
      size={12}
      style={
        Object {
          "color": undefined,
          "marginHorizontal": 5,
        }
      }
    />
  </View>
</TouchableOpacity>
`;
